app/src/main/java/net/squanchy/schedule/domain/view/Schedule.kt

```diff
@@ -1,12 +1,50 @@
 package net.squanchy.schedule.domain.view
 
-data class Schedule(val pages: List<SchedulePage>) {
+import net.squanchy.support.system.CurrentTime
+import org.joda.time.DateTimeZone
+
+private const val NOT_FOUND_INDEX = -1
+private const val FIRST_PAGE_INDEX = 0
+
+private const val CURRENT_SLOT_THRESHOLD = .6f
+
+data class Schedule(val pages: List<SchedulePage>, val timezone: DateTimeZone) {
 
     val isEmpty: Boolean
         get() = pages.isEmpty()
 
     companion object {
 
-        fun create(pages: List<SchedulePage>) = Schedule(pages)
+        fun create(pages: List<SchedulePage>, timezone: DateTimeZone) = Schedule(pages, timezone)
     }
+
+    fun findTodayIndexOrDefault(currentTime: CurrentTime) =
+        pages
+            .indexOfFirst { page ->
+                val now = currentTime.currentDateTime().toDateTime(timezone)
```

Comment:
here, extracting outside of the lambda (inside the method) so you don't create a date per page